config.go

```diff
@@ -21,8 +25,8 @@ type Spec struct {
 type Process struct {
 	// Terminal creates an interactive terminal for the container.
 	Terminal bool `json:"terminal"`
-	// User specifies user information for the process.
-	User User `json:"user"`
+	// User specifies user information for the process. It is a platform-specific structure.
+	User json.RawMessage `json:"user"`
```

Comment:
On Wed, Dec 16, 2015 at 10:25:08AM -0800, Vish Kannan wrote:
> > @@ -21,8 +25,8 @@ type Spec struct {
> >  type Process struct {
> >  	// Terminal creates an interactive terminal for the container.
> >  	Terminal bool `json:"terminal"`
> > -	// User specifies user information for the process.
> > -	User User `json:"user"`
> > +	// User specifies user information for the process. It is a platform-specific structure.
> > +	User json.RawMessage `json:"user"`
> 
> Can `User` be a string?

User as a string requires /etc/passwd lookups or some such on Linux,
and I don't think we want to *require* that (although I'm fine with it
as an option).  See my suggestions in #191 for an alternative that
allows both name-based and number-based identification [1], but that's
going to be platform-specific.

[1]: https://github.com/opencontainers/specs/pull/191#issuecomment-141150387